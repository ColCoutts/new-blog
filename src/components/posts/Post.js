import React from 'react';
import PropTypes from 'prop-types';

function Post({ post, deletePost }) {
  return (
    <section>
      <p>{post.title}</p>
      <a href='#' onClick={deletePost.bind(null, post.id)}>&#x1c2;</a>
    </section>
  );
}

Post.propTypes = {
  deletePost: PropTypes.func.isRequired,
  post: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired
  }).isRequired
};

export default Post;
