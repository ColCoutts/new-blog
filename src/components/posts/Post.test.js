import React from 'react';
import { shallow } from 'enzyme';
import Post from './Post';

describe('post snapshot test', () => {
  it('renders Post snapshot', () => {
    const newPost = {
      title: 'hi',
      body: 'test'
    };
    const wrapper = shallow(<Post post={newPost} deletePost={ () => {} } />);
    expect(wrapper).toMatchSnapshot();
  });
});
