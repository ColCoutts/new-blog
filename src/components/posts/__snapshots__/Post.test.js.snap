// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post snapshot test renders Post snapshot 1`] = `
<section>
  <Link
    to="/posts/undefined"
  >
    hi
  </Link>
  <a
    href="#"
    onClick={[Function]}
  >
    Ç‚
  </a>
</section>
`;
